CMakeLists.txt

```diff
@@ -219,13 +220,65 @@ if (NOT ${MODULE} STREQUAL "casa")
         endif (NOT ${MODULE} STREQUAL "measures")
     endif (NOT ${MODULE} STREQUAL "tables")
 endif (NOT ${MODULE} STREQUAL "casa")
+
 if (BUILD_PYTHON)
-    set (_modules ${_modules} python)
-    find_package(Boost REQUIRED COMPONENTS python)
+    message("Looking for python2 specific environment...")
+    set(Python_FIND_VERSION 2)
+    set(PythonInterp_FIND_VERSION_MAJOR 2)
     find_package(Python REQUIRED)
-    find_package (NUMPY REQUIRED)
+    if (PYTHONINTERP_FOUND)
+        find_package(Boost REQUIRED COMPONENTS python)
+        find_package (NUMPY REQUIRED)
+    endif(PYTHONINTERP_FOUND)
+    set(PYTHON2_EXECUTABLE ${PYTHON_EXECUTABLE})
+    set(PYTHON2_LIBRARIES ${PYTHON_LIBRARIES})
+    set(PYTHON2_NUMPY_INCLUDE_DIRS ${NUMPY_INCLUDE_DIRS})
+    set(PYTHON2_Boost_LIBRARIES ${Boost_LIBRARIES})
+    set(PYTHON2_Boost_INCLUDE_DIRS ${Boost_INCLUDE_DIRS})
+    set(PYTHON2_Boost_PYTHON_LIBRARY_RELEASE ${Boost_PYTHON_LIBRARY_RELEASE})
+    set(PYTHON2_INCLUDE_DIRS ${PYTHON_INCLUDE_DIRS})
+    set (_modules ${_modules} python)
 endif (BUILD_PYTHON)
 
+if (BUILD_PYTHON3)
+    message("Looking for python3 specific environment...")
+
+    # unset all variables set by find python so we can do it again for python3
+    unset(PYTHONINTERP_FOUND CACHE)
+    unset(PYTHONLIBS_FOUND CACHE)
+    unset(PYTHON_EXECUTABLE CACHE)
+    unset(PYTHON_FOUND CACHE)
+    unset(_Python_NAMES CACHE)
+    unset(PYTHON_LIBRARY)
+    unset(PYTHON_LIBRARY CACHE)
+    unset(PYTHON_LIBRARIES)
+    unset(PYTHON_LIBRARY_RELEASE)
+    unset(PYTHON_INCLUDE_DIR)
+    unset(PYTHON_INCLUDE_DIR CACHE)
```

Comment:
hm, I forgot that this may need some improvements. Now you cannot set a custom location for a python interpreter.